### PR TITLE
fix: Composer prevails over env for Nginx and Composer too

### PR DESCRIPTION
**Pull request by Frzk**
Add tests to check that:

- The default Nginx version is installed if nothing is specified (this version is different in `scalingo-20` and `scalingo-22`).
- The user can specify an Nginx version by using `NGINX_VERSION`
- The user can specify an Nginx version by using `composer.json`
- Using `composer.json` will always prevail over `NGINX_VERSION` if both are set.
- 
- The default Composer version is installed if the app is not a Classic PHP app.
- The user can specify a Composer version by using `NGINX_VERSION`
- The user can specify a Composer version by using `composer.json`
- Using `composer.json` will always prevail over `NGINX_VERSION` if both are set.

I had to make quite a few changes to be able to do that in a way that follows the testing logic we have.

### CODE REVIEW

File: test/helpers [hunk 1]
```diff
@@ -2,9 +2,50 @@
...
+    # Test PHP default modules:
+    test::helpers::default_php_modules
+
+    # Test Nginx version:
+    test::helpers::nginx "${nginx_version}"
+
+    # No Composer
+}
+
+test::helpers::composer_deploy() {
```
Comment (EtienneM) [inline test/helpers:46]: suggestion(non-blocking): with both `composer_deploy` and `classic_deploy` sharing most of the code base, we could create a common function named `test::helpers::deploy_common` that is called by these functions.

File: test/helpers [hunk 2]
```diff
@@ -54,7 +111,17 @@ test::helpers::detect() {
 test::helpers::compile() {
     # Runs the `bin/compile` script of the buildpack and then:
     # - Asserts that is succeeded
-    # - Asserts that the version of installed PHP is the one expected
+
+    test::utils::compile
+    # We can't use assertCapturedSuccess here:
+    # With an empty composer.json file, composer will use stderr to warn us
+    # that nothing had to be installed, causing the test to fail for no
+    # reason -_-
```
Comment (EtienneM) [inline test/helpers:119]: praise: thanks for the comment :+1:

File: test/helpers [hunk 3]
```diff
@@ -74,49 +141,54 @@ test::helpers::compile() {
...
+    #
+    # Arguments:
+    #   $1 > $version: Expected Nginx version
+
+    local version="${1}"
 
-    test::utils::capture ./vendor/php/bin/php --modules
+    test::utils::capture nginx -version
+    # For some reason known to Nginx, `-version` outputs on stderr...
+    # See https://trac.nginx.org/test/ticket/592 for further details.
```
Comment (EtienneM) [inline test/helpers:184]: praise: wow nicely found!

File: test/run [hunk 4]
```diff
@@ -29,22 +29,41 @@ setUp() {
...
+    # We'll probably update PATH during the test,
+    # let's keep the original value safe
+    PREV_PATH="${PATH}"
+    export PREV_PATH
+
     # Copy the buildpack code into BUILDPACK_DIR:
     cp --archive "$( pwd )"/* "${BUILDPACK_DIR}/"
+
+    # Make sure we have a /app directory:
+    mkdir -p "/app"
```
Comment (EtienneM) [inline test/run:41]: suggestion(non-blocking): I agree that long options are better
```suggestion
    mkdir --parents "/app"
```

**Review by EtienneM — APPROVED** [review]
a couple of suggestions, but already approving the PR

File: test/helpers (hunk 1 above)
Comment (Frzk) [reply test/helpers:68]: Right, applied in https://github.com/Scalingo/php-buildpack/pull/409/commits/4217951328756a3b8507d264f1207058369429db

**Review by EtienneM — APPROVED** [review]

**Pull request merged**